### ISSUE TITLE
Difficulty saving graphs from browser

### ISSUE THREAD

**Issue by eschoeller** [Cacti/cacti#413]
In the past I have relied heavily on right clicking any graph and using the browser functions to perform numerous functions, such as 'copy link location', 'save image', etc. In 1.0.4 right clicking anywhere on an RRD graph brings up a nifty little zoom menu (cool!) but I'm left with very few options now to actually save graphs quickly. 
I found I can select "Time Graph View" and now right click on an image again, but I'm limited to the view options defined for that data profile. 
Would it be possible to limit the zoom menu to only the graph area, but leave the rest of the RRD area (legend, axis, etc) back to the browser to control?

**Comment by eschoeller** [comment]
It would also be nice to drag and drop graphs directly to other applications, such as an e-mail client. This never worked for me in previous versions but I was just surprised to see that I pulled it off from dragging an image from the "Time Graph View" into an e-mail. Man that would save huge amounts of time for me!

**Comment by cigamit** [comment]
Assigning this to Browniebraun.  This is his area of expertise.

**Comment by browniebraun** [comment]
With 1.0.5 you should be able to download the graph to your local disk. (Zoom Menu -> Save Image)
I can commit later today my latest changes to offer you the option to open the graph within a new tab or window (browser dependent) with a real URL. Currently I'm have some compatibility issues with another option to copy that link, but I will to try to get that working and check if it is possible to support drag and drop.

Main reason why the old browser context menu can not be used anymore is the fact that we are using embedded and base64 encoded images loaded by using Ajax. This avoids that the browser will deliver you a correct graph URL that you can share with other users. The browser will return the base64 code only in that case. Same situation if you would like to open the graph in another tab / window.

Regards
-Andi

**Comment by browniebraun** [comment]
Please give my latest commitment a try: https://github.com/Cacti/cacti/commit/befd64c427471ea0007f6328c88b9759c48ef04d

So far I've no solution for drag and drop as well as a "copy image to clipboard" function.

**Comment by eschoeller** [comment]
Pulled the latest from git today and yes the zoom menu has those three options working just great. I figured I would be able to open the graph in a new tab, then drag and drop from there, but for some reason that doesn't work. I'll try from my machine at work sometime, it has a slightly new O/S which could be making a difference.

**Comment by cigamit** [comment]
With Chrome, when I select "Open in New Tab", on the new Tab I can now "Copy Image" which then copies it to the clipboard.  I can also Drag and Drop it from there into my Email client (Zimbra) without any issues.

That should cover all your use cases.  Marking this as closed until we hear otherwise.
